core/arch/arm/include/ffa.h

```diff
@@ -0,0 +1,85 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright (c) 2020, Linaro Limited
+ * Copyright (c) 2018-2019, Arm Limited. All rights reserved.
+ */
+
+#ifndef __FFA_H
+#define __FFA_H
+
+/* This is based on the FF-A 1.0 EAC specification */
+
+#include <smccc.h>
+#include <stdint.h>
+
+/* Error codes */
+#define FFA_NOT_SUPPORTED	-1
+#define FFA_INVALID_PARAMETERS	-2
+#define FFA_NO_MEMORY		-3
+#define FFA_BUSY		-4
+#define FFA_INTERRUPTED		-5
+#define FFA_DENIED		-6
+#define FFA_RETRY		-7
+
+/* FFA_VERSION helpers */
+#define FFA_VERSION_MAJOR		U(1)
+#define FFA_VERSION_MAJOR_SHIFT		16
+#define FFA_VERSION_MAJOR_MASK		U(0x7FFF)
+#define FFA_VERSION_MINOR		U(0)
+#define FFA_VERSION_MINOR_SHIFT		0
+#define FFA_VERSION_MINOR_MASK		U(0xFFFF)
```

Comment:
GENMASK_32(15, 0)

Reply:
These simple masks are in my opinion easier to read if written plainly as here instead of using `GENMASK_32()`. I'll change it if you think it's important.

Reply:
no strong opinion on this. Keep 0xFFFF if you think it's easier to read and maintain.